utils/test/unit/test_utils.py

```diff
@@ -0,0 +1,33 @@
+import logging
```

Comment:
I don't like the idea of copying the same utils file for every test type. I'm not saying this should be part of this PR, but I think that common test utils should be extracted to a separate package and reused. Now we have it four times.

Reply:
The reason we have it multiple times is these are just sample tests so we have something running rather than being utility tests. I understand it's a bit strange but there's nothing else we can really test. I agree that the last function should be abstracted and not held within the same files as tests though.

Reply:
Abstracted the function. Leaving the rest the same as they are just example tests in the pipelines, and will be changed as the repo is used.

Reply:
looks good